src/wm2/src/wm2_radio.c

```diff
@@ -1111,6 +1111,7 @@ wm2_rconf_recalc(const char *ifname, bool force)
             LOGD("%s: ignoring rconf channel %d: no vifs available yet",
                   rconf.if_name, rconf.channel);
             rconf.channel = rstate.channel;
+            rconf.channel_exists = rstate.channel_exists;
```

Comment:
@stschake Hmm.. I agree. Good catch! The `vif_configs_len==0` condition typically doesn't last long, so the bug went unnoticed.

With that said, the entire condition should be reworked to avoid confusion, like so perhaps:
```
if (want && rconf.vif_configs_len == 0) {
   LOGD(...);
   rconf.channel_exists = false;
   rconf.ht_mode_exists = false;
   rconf.hw_mode_exists = false;
}
```

Actual values need not be copied since all comparators first check if the conf side has `_exists == true`.

It'd be great if you can extend the commit log a bit more: with some background, conditions for reproduction and impact of not fixing it.

Can you re-spin (amend + force-push) with that in mind, please?

Reply:
Thanks for the review and suggestion, setting only _exists is a much more elegant solution. I've amended the PR.